File: remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        if(head==null || head.next==null)
            return null;
        ListNode dummy = new ListNode(-1);
        dummy.next=head;
        ListNode start = dummy;
        ListNode end = dummy;
        for(int i=0;i<=n;i++)
        {
            end = end.next;
        }
        
        while(end!=null)
        {
            start = start.next;
            end = end.next;
        }
        ListNode extra = start.next;
        start.next = start.next.next;
        if(extra!=null)
        {
            extra.next=null;
        }
        return dummy.next;
    }
}

File: java/RemoveNthNodeFromEndofList.java
// Time Complexity : O(N)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : no


// Your code here along with comments explaining your approach
2-pass solution
Get the length of the ListNode and subtract the n from it. Traverse until then and change the pointer to next next.
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode slow = head;
        int count = 0;
        while(slow!=null){
            count= count+1;
            slow = slow.next;
        }
        slow = dummy;
        System.out.println(count);
        count = count-n;
        while(count>0){
            count--;
            slow = slow.next;
        }
        slow.next = slow.next.next;
        return dummy.next;
    }
}